reactor-core/src/main/java/reactor/core/publisher/Sinks.java

```diff
@@ -600,6 +600,17 @@ public Emission getReason() {
 		@Deprecated
 		void emitError(Throwable error);
 
+		/**
+		 * Determines if the sink currently has at least one subscriber.
+		 * <p>
+		 * This is a best effort peek at the sink state, and a subsequent attempt at emitting
+		 * to the sink might still return {@link Emission#FAIL_ZERO_SUBSCRIBER} where relevant
+		 * (generally in {@link #tryEmitNext(Object)}).
+		 *
+		 * @return {@code true} if the sink has at least one subscriber at the time of invocation, {@code false} otherwise
+		 */
+		boolean hasSubscriber();
```

Comment:
WDYT about using `isSubscribed()`, to avoid singular vs plural confusion?

Reply:
I think it could introduce even more cognitive dissonance since it can either be read as "the sink `is subscribed` **to** something" or "the sink `is subscribed` _to_ **by** something" 🤔

Reply:
On a side note I elected to use singular rather than plural form because I find plural form to be more easily interpreted as "strictly greater than one" semantic, while singular can be better accepted as meaning "at least one". Plus plural form would feel weird on `unicast` variants.

Reply:
if we follow this path, the same applies to "the sink has a subscriber that it created" vs "there is a subscriber and this sink has it" :D in fact, the latter is even more confusing IMO :)

Reply:
Some further thoughts. The method could return the number of current subscribers, if that extra information has potential to be useful? Also it would be nice for the the name to include something like "current" or "active" to indicate the transient nature.

Reply:
done, with `activeSubscriberCount()`